frontend/src/services/PromotionService.es6

```diff
@@ -58,17 +58,17 @@ export default class {
             method: 'POST',
             url: '/promotion/validate',
             data: promotion
-        }).then(r => {
```

Comment:
The library only treats 2XX as successes, so if we use status codes correctly we can avoid looking at the JSON here: https://docs.angularjs.org/api/ng/service/$http#$http-returns